### PR TITLE
Unhide overlapping grouped bars within trace

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/3651 - where I decided to not add another attribute as the problem described in https://github.com/plotly/plotly.js/issues/3651 only affects bar graphs with `barmode: 'group'`. See https://github.com/plotly/plotly.js/issues/3651#issuecomment-476356081 for more on the topic.

cc @plotly/plotly_js @nicolaskruchten

### CODE REVIEW

File: src/traces/bar/cross_trace_calc.js [hunk 1]
```diff
@@ -57,52 +57,57 @@ function setGroupPositions(gd, pa, sa, calcTraces) {
...
+            // exclude from the group those traces for which the user set an offset
+            excluded = [];
+            included = [];
+            for(i = 0; i < calcTraces.length; i++) {
+                calcTrace = calcTraces[i];
+                fullTrace = calcTrace[0].trace;
+
+                if(fullTrace.offset === undefined) included.push(calcTrace);
+                else excluded.push(calcTrace);
+            }
```
Comment (archmoj) [inline src/traces/bar/cross_trace_calc.js]: Could we put this logic in a function and reuse it for the case of `relative` below?

File: src/traces/bar/cross_trace_calc.js [hunk 2]
```diff
@@ -201,12 +214,12 @@ function setGroupPositionsInGroupMode(gd, pa, sa, calcTraces) {
...
-    var stack = barmode === 'stack';
-    var relative = barmode === 'relative';
     var barnorm = fullLayout.barnorm;
-    var separateNegativeValues = relative;
-    var dontMergeOverlappingData = !(barnorm || stack || relative);
-    var sieve = new Sieve(calcTraces, separateNegativeValues, dontMergeOverlappingData);
+
+    var sieve = new Sieve(calcTraces, {
+        separateNegativeValues: barmode === 'relative',
+        dontMergeOverlappingData: !(barnorm || barmode === 'stack' || barmode === 'relative')
```
Comment (archmoj) [inline src/traces/bar/cross_trace_calc.js]: The function now has a more clear interface block which is great. 
Now that the keys would be presented in the `.min` files could we zip `separateNegativeValues` into e.g. `separateNegVals` and `dontMergeOverlappingData` into a short form?

**Comment by etpinard** [comment]
Thanks for the review @archmoj - I'll until the `waterfall` PR is merged before addressing the comments.

File: src/traces/bar/cross_trace_calc.js (hunk 2 above)
Comment (etpinard) [reply src/traces/bar/cross_trace_calc.js]: Done in c019c5ca8

File: src/traces/bar/cross_trace_calc.js (hunk 1 above)
Comment (etpinard) [reply src/traces/bar/cross_trace_calc.js]: I'll pass if you don't mind. That would a pretty ugly function (we would have to pass functions `setGroupPositionsInGroupMode`, etc) and only save a few lines.

**Pull request merged**